hydra-node/src/Hydra/Network/Ouroboros.hs

```diff
@@ -122,10 +128,10 @@ withOuroborosNetwork ::
   (ToCBOR outbound, FromCBOR outbound) =>
   (ToCBOR inbound, FromCBOR inbound) =>
   Tracer IO (WithHost (TraceOuroborosNetwork outbound)) ->
-  Host ->
-  [Host] ->
+  HydraNetworkConfig ->
+  (HydraHandshakeRefused -> IO ()) ->
```

Comment:
Must: not make the same mistake again that we refactored

Having an additional callback is exactly what we removed one PR prior to this work and a separate callback here, will make handling this "more optional" to calling code. Instead the `inbound` type should be enhanced by the case which can happen.

Reply:
We investigated and agreed that extending the `inbound` would make handshake failure also authenticated and "reliable" messages .. which is not what we likely want. So it's fine to have a second callback here.